File: Month.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WpfApplication1
{
    class Month
    {
        //Fields:
        string hebName;
        string engName;
        public ArrayList daysArr;
   
        int headF;
        int lengthF;
        int zorderF;

        //Properties:
        public string hName { get { return hebName; } set { hebName = value; } }
        public string eName { get { return engName; } set { engName = value; } }
        public int head { get { return headF; } set { headF = value; } }
        public int length { get { return lengthF; } set { lengthF = value; } }
        public int zorder { get { return zorderF; } set { zorderF = value; } }
 
        //C'tors:
 
        public Month()
          : this("", "")
        {
        }

        public Month(string engName, string hebName)
        {
            hName = hebName;
            eName = engName;
            daysArr = new ArrayList();
        }

        //Methods:
        public void AddDay(Day newDay)
        {
            daysArr.Add(newDay);
        }
    }

    class hMonth : Month
    {
        //Fields:
        string gregorianName;
        public string gName { get { return gregorianName; } set { gregorianName = value; } }

        public hMonth(string engName, string hebName)
            : base(engName, hebName)
        {

        }

    }
    class gMonth : Month
    {
        //Fields:
        string shebName;
        string sengName;
        public string shName { get { return shebName; } set { shebName = value; } }
        public string seName { get { return sengName; } set { sengName = value; } }

        public gMonth(string engName, string hebName)
            : base(engName, hebName)
        {
          
        }    

        public gMonth(string shortHebName, string shortEngName, string hebName, string engName)
            : base(engName, hebName)
        {
            shName = shortHebName;
            seName = shortEngName;
        }

        //Methods:
        public void SetShortNames(string shortEngName, string shortHebName)
        {
            shName = shortHebName;
            seName = shortEngName;
            return;
        }
    }


}

File: Month.js
var myNS = myNS || {};

myNS.Month = function (eName, hName ) {
  // private members
  var _daysArr=[];
  var _hebName = hName;
  var _engName = eName;
   
	var _head=0;
	var _length=0;
	var _zorder=0;
	
  var _AddDay = function (newDay) {
    _daysArr.push(newDay);
  };
  
  return { // public parts (aka interface)
    hebName : _hebName,
    engName : _engName,
  	daysArr : _daysArr,
  	head : _head,
  	Mlength : _length,
  	zorder : _zorder,
    AddDay : _AddDay
  };
};

myNS.gMonth = function (seName, shName, eName, hName ) { // Inherits Month
 
    //Fields:
    var _shortHebName=shName;
    var _shortEngName=seName;

    //the next code inherits DAY ...
    var SuperMonth = new myNS.Month(eName, hName);
    this.prototype = SuperMonth.prototype;
    for (var i in SuperMonth) {
        this[i] = SuperMonth[i];
    }
    
      return { // public parts (aka interface)
        hebName : SuperMonth.hebName,
        engName : SuperMonth.engName,
        daysArr : SuperMonth.daysArr,
        head : SuperMonth.head,
        Mlength : SuperMonth.Mlength,
        zorder : SuperMonth.zorder,
        AddDay : SuperMonth.AddDay,
        shortHebName : _shortHebName,
        shortEngName : _shortEngName
      };

}

//---------------------------------------------------------------------------
